### PR TITLE
[Android] Refactor the `flutter run` Android console output test

### PR DESCRIPTION
**Pull request by loic-sharma**
The [`run_release_test`](https://github.com/flutter/flutter/blob/master/dev/devicelab/bin/tasks/run_release_test.dart) integration test verifies that that the console output of `flutter run -d android --release` is expected. In https://github.com/flutter/flutter/pull/114645, a `RunOutputTask` abstraction was introduced so that the test's logic can be reused by other platforms like macOS and Windows. This moves the `run_release_test` to that new `RunOutputTask` abstraction.

Part of https://github.com/flutter/flutter/issues/111577

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/devicelab/bin/tasks/run_release_test.dart [hunk 1]
```diff
@@ -2,161 +2,9 @@
...
-      })..stderr
-        .transform<String>(utf8.decoder)
-        .transform<String>(const LineSplitter())
-        .listen((String line) {
-        print('uninstall:stderr: $line');
-        stderr.add(line);
-      });
-      if (await uninstall.exitCode != 0) {
-        throw 'flutter install --uninstall-only failed.';
-      }
```
Comment (loic-sharma) [inline dev/devicelab/bin/tasks/run_release_test.dart:44]: This was moved to `AndroidRunOutputTest.prepare`

File: dev/devicelab/bin/tasks/run_release_test.dart [hunk 2]
```diff
@@ -2,161 +2,9 @@
...
-            !line.startsWith('Building flutter tool...') &&
-            !line.startsWith('Running "flutter pub get" in ui...') &&
-            !line.startsWith('Initializing gradle...') &&
-            !line.contains('settings_aar.gradle') &&
-            !line.startsWith('Resolving dependencies...') &&
-            // Catch engine piped output from unrelated concurrent Flutter apps
-            !line.contains(RegExp(r'[A-Z]\/flutter \([0-9]+\):')) &&
-            // Empty lines could be due to the progress spinner breaking up.
-            line.length > 1
-          ) {
```
Comment (loic-sharma) [inline dev/devicelab/bin/tasks/run_release_test.dart:68]: I removed these filters as they weren't necessary. This test ignores unmatched lines: it only checks that the expected lines are in standard output in the correct order.

File: dev/devicelab/bin/tasks/run_release_test.dart [hunk 3]
```diff
@@ -2,161 +2,9 @@
...
-          }
-          if (line.contains('Quit (terminate the application on the device).')) {
-            ready.complete();
-          }
-        });
-      run.stderr
-        .transform<String>(utf8.decoder)
-        .transform<String>(const LineSplitter())
-        // TODO(egarciad): Remove once https://github.com/flutter/flutter/issues/95131 is fixed.
-        .skipWhile((String line) => line.contains('Mapping new ns'))
```
Comment (loic-sharma) [inline dev/devicelab/bin/tasks/run_release_test.dart:79]: I removed this as this is no longer necessary.

File: dev/devicelab/lib/tasks/run_tests.dart [hunk 4]
```diff
@@ -11,21 +11,104 @@ import '../framework/framework.dart';
...
+}
+
 TaskFunction createMacOSRunReleaseTest() {
   return DesktopRunOutputTest(
     // TODO(cbracken): https://github.com/flutter/flutter/issues/87508#issuecomment-1043753201
     // Switch to dev/integration_tests/ui once we have CocoaPods working on M1 Macs.
     '${flutterDirectory.path}/examples/hello_world',
     'lib/main.dart',
     release: true,
+    allowStderr: true,
```
Comment (loic-sharma) [inline dev/devicelab/lib/tasks/run_tests.dart:25]: The macOS `flutter run` test has stderr output from xcode build. The original test from before this refactoring also [ignored stderr output](https://github.com/flutter/flutter/blob/7020f59ace273d8644f0c8f3d0c6e7c9c95ba51a/dev/devicelab/bin/tasks/run_release_test_macos.dart#L74).

Comment (jmagman) [reply dev/devicelab/lib/tasks/run_tests.dart:25]: @stuartmorgan @cbracken do you remember which issue is tracking cleaning up the macOS Xcode -> stderr mapping?  I can't find it at the moment...

File: dev/devicelab/bin/tasks/run_release_test.dart (hunk 3 above)
Comment (loic-sharma) [reply dev/devicelab/bin/tasks/run_release_test.dart:79]: Re-added as it turns out these are still outputted

File: dev/devicelab/lib/tasks/run_tests.dart [hunk 5]
```diff
@@ -11,24 +11,121 @@ import '../framework/framework.dart';
...
+  AndroidRunOutputTest({required super.release}) : super(
+    '${flutterDirectory.path}/dev/integration_tests/ui',
+    'lib/main.dart',
+  );
+
+  @override
+  Future<void> prepare(String deviceId) async {
+    // Uninstall if the app is already installed on the device to get to a clean state.
+    final List<String> stderr = <String>[];
+    print('uninstalling...');
```
Comment (a-wallen) [inline dev/devicelab/lib/tasks/run_tests.dart:39]: Are print statements allowed here?

Comment (loic-sharma) [reply dev/devicelab/lib/tasks/run_tests.dart:39]: AFAICT yes, there's lots of `print`s throughout the devicelab tasks: https://github.com/search?q=repo%3Aflutter%2Fflutter+path%3Adev%2Fdevicelab%2Flib%2Ftasks+print&type=code

File: dev/devicelab/lib/tasks/run_tests.dart (hunk 4 above)
Comment (stuartmorgan) [reply dev/devicelab/lib/tasks/run_tests.dart:25]: I don't remember if we actually filed anything for that specifically.

**Review by GaryQian — COMMENTED** [review]
LGTM!

**Review by GaryQian — APPROVED** [review]

**Pull request merged**